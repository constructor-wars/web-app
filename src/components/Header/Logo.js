import React from 'react';
import Logoimg from '../../images/constructor-labs-badge.png';

export default function Logo() {
  return (
    <div>
      <img className="logoimg" src={Logoimg} />
    </div>
  );
}
