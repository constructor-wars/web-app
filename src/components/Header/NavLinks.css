.nav_links {
  margin: auto 30px auto auto;
}

.nav_buttons {
  color: var(--light);
  height: 25px;
  margin: 20px 0px;
  width: 100px;
  align-self: center;
  background-color: var(--dark);
  cursor: pointer;
}
