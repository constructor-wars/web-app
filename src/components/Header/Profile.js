import React from 'react';
import { Link } from 'react-router-dom';

export default function Profile() {
  return (
    <div className="profile">
      <Link to="/profile">Profile</Link>
    </div>
  );
}
