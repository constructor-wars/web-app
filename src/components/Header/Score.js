import React from 'react';

export default function Score() {
  return <div className="score">Score</div>;
}
