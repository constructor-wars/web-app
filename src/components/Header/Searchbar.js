import React from 'react';

export default function Searchbar() {
  return (
    <div>
      <input className="searchbar" />
    </div>
  );
}
