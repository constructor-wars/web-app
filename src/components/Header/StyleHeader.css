.background {
  display: flex;
  flex-direction: row;
  background-color: var(--background__color);
  margin: 0 0;
  padding: 10px;
  min-height: 100px;
}

.searchbar {
  width: 250px;
  margin-left: 100px;
}

.profile,
.score {
  margin: 0 150px;
}

.logoimg {
  height: 50px;
  width: 50px;
}
/* .score {
  margin-left: auto;
} */
