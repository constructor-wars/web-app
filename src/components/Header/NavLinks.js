import React from "react";
import { Link } from "react-router-dom";
import "./NavLinks.css";

import { FiPlay, FiBookOpen, FiLogOut } from "react-icons/fi";
import { GoInfo, GoGear, GoDashboard } from "react-icons/go";

export function NavLinks() {
  return (
    <div className="nav_links">
      <Link to="/editor">
        <div className="nav_buttons">
          <FiPlay className="nav_links__icon" /> Start Coding
        </div>
      </Link>
      <Link to="/dashboard">
        <div className="nav_buttons">
          <GoDashboard className="nav_links__icon" /> Dashboard
        </div>
      </Link>
      <Link to="/syllabus">
        <div className="nav_buttons">
          {" "}
          <FiBookOpen className="nav_links__icon" /> Syllabus
        </div>
      </Link>
      <Link to="/about">
        <div className="nav_buttons">
          {" "}
          <GoInfo className="nav_links__icon" /> About
        </div>
      </Link>
      <a href="/logout">
        <div className="nav_buttons">
          {" "}
          <FiLogOut className="nav_links__icon" /> Logout
        </div>
      </a>
      <Link to="/admin">
        <div className="nav_buttons">
          {" "}
          <GoGear className="nav_links__icon" /> Admin
        </div>
      </Link>
    </div>
  );
}
