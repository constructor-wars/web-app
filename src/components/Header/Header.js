import React, { Component } from 'react';
import './StyleHeader.css';
import Logo from './Logo';
import Searchbar from './Searchbar';
import Score from './Score';
import Profile from './Profile';

export default class Header extends Component {
  render() {
    return (
      <header>
        <div className="background">
          <Logo className="logo" />
          <Searchbar className="searchbar" />
          <Score />
          <Profile />
        </div>
      </header>
    );
  }
}
