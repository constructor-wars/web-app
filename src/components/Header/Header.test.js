import React from "react";
import Header from "./Header";
import renderer from "react-test-renderer";
import { shallow, render } from "enzyme"; // import shallow renderer from enzyme

describe("Header", () => {
  test("matches the snapshot", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render has Score", () => {
    const wrapper = render(<Header />);
    const text = wrapper.find(".score").text();
    expect(text).toEqual("Score");
  });
});
