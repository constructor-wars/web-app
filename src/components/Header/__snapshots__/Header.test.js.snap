// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<header>
  <div
    className="background"
  >
    <div>
      <a
        href="/"
      >
        <img
          className="logoimg"
          src="test-file-stub"
        />
      </a>
    </div>
    <div
      className="nav_links"
    >
      <a
        href="/editor"
        onClick={[Function]}
      >
        <div
          className="nav_buttons"
        >
          <svg
            attr={
              Object {
                "fill": "none",
                "stroke": "currentColor",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": "2",
                "viewBox": "0 0 24 24",
              }
            }
            className="undefined nav_links__icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
          >
            <polygon
              points="5 3 19 12 5 21 5 3"
            />
          </svg>
           Start Coding
        </div>
      </a>
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        <div
          className="nav_buttons"
        >
          <svg
            attr={
              Object {
                "viewBox": "0 0 16 16",
              }
            }
            className="undefined nav_links__icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M9 5H8V4h1v1zm4 3h-1v1h1V8zM6 5H5v1h1V5zM5 8H4v1h1V8zm11-5.5l-.5-.5L9 7c-.06-.02-1 0-1 0-.55 0-1 .45-1 1v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-.92l6-5.58zm-1.59 4.09c.19.61.3 1.25.3 1.91 0 3.42-2.78 6.2-6.2 6.2-3.42 0-6.21-2.78-6.21-6.2 0-3.42 2.78-6.2 6.2-6.2 1.2 0 2.31.34 3.27.94l.94-.94A7.459 7.459 0 0 0 8.51 1C4.36 1 1 4.36 1 8.5 1 12.64 4.36 16 8.5 16c4.14 0 7.5-3.36 7.5-7.5 0-1.03-.2-2.02-.59-2.91l-1 1z"
              fillRule="evenodd"
            />
          </svg>
           Dashboard
        </div>
      </a>
      <a
        href="/syllabus"
        onClick={[Function]}
      >
        <div
          className="nav_buttons"
        >
           
          <svg
            attr={
              Object {
                "fill": "none",
                "stroke": "currentColor",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": "2",
                "viewBox": "0 0 24 24",
              }
            }
            className="undefined nav_links__icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M2 3h6a4 4 0 0 1 4 4v14a3 3 0 0 0-3-3H2z"
            />
            <path
              d="M22 3h-6a4 4 0 0 0-4 4v14a3 3 0 0 1 3-3h7z"
            />
          </svg>
           Syllabus
        </div>
      </a>
      <a
        href="/about"
        onClick={[Function]}
      >
        <div
          className="nav_buttons"
        >
           
          <svg
            attr={
              Object {
                "viewBox": "0 0 14 16",
              }
            }
            className="undefined nav_links__icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 14 16"
            width="1em"
          >
            <path
              d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"
              fillRule="evenodd"
            />
          </svg>
           About
        </div>
      </a>
      <a
        href="/logout"
      >
        <div
          className="nav_buttons"
        >
           
          <svg
            attr={
              Object {
                "fill": "none",
                "stroke": "currentColor",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": "2",
                "viewBox": "0 0 24 24",
              }
            }
            className="undefined nav_links__icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
            />
            <polyline
              points="16 17 21 12 16 7"
            />
            <line
              x1="21"
              x2="9"
              y1="12"
              y2="12"
            />
          </svg>
           Logout
        </div>
      </a>
      <a
        href="/admin"
        onClick={[Function]}
      >
        <div
          className="nav_buttons"
        >
           
          <svg
            attr={
              Object {
                "viewBox": "0 0 14 16",
              }
            }
            className="undefined nav_links__icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 14 16"
            width="1em"
          >
            <path
              d="M14 8.77v-1.6l-1.94-.64-.45-1.09.88-1.84-1.13-1.13-1.81.91-1.09-.45-.69-1.92h-1.6l-.63 1.94-1.11.45-1.84-.88-1.13 1.13.91 1.81-.45 1.09L0 7.23v1.59l1.94.64.45 1.09-.88 1.84 1.13 1.13 1.81-.91 1.09.45.69 1.92h1.59l.63-1.94 1.11-.45 1.84.88 1.13-1.13-.92-1.81.47-1.09L14 8.75v.02zM7 11c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
              fillRule="evenodd"
            />
          </svg>
           Admin
        </div>
      </a>
    </div>
    <a
      href="/profile"
      onClick={[Function]}
    >
      <div
        className="navbarprofile_container"
      >
        <img
          alt="bob 3"
          className="navbarprofile_picture"
          src="bob4"
        />
      </div>
    </a>
  </div>
</header>
`;
